File: android/src/main/java/io/lokal/notifications/LocalNotificationReceiver.java
package io.lokal.notifications;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.app.NotificationManager;
import android.os.Bundle;
import androidx.core.app.NotificationCompat;

public class LocalNotificationReceiver extends BroadcastReceiver {
    
    @Override
    public void onReceive(Context context, Intent intent) {
        try {
            // Extract notification data from intent
            int notificationId = intent.getIntExtra("notification_id", 0);
            String title = intent.getStringExtra("title");
            String body = intent.getStringExtra("body");
            long scheduledTime = intent.getLongExtra("scheduled_time", 0);
            Bundle data = intent.getBundleExtra("data");
            
            // Create and show the notification
            NotificationManager notificationManager = (NotificationManager) context.getSystemService(Context.NOTIFICATION_SERVICE);
            
            NotificationCompat.Builder builder = new NotificationCompat.Builder(context, "LocalNotifications")
                .setContentTitle(title)
                .setContentText(body)
                .setSmallIcon(getNotificationIcon(context))
                .setPriority(NotificationCompat.PRIORITY_DEFAULT)
                .setAutoCancel(true)
                .setWhen(scheduledTime);
            
            // Add any extra data to the notification
            if (data != null) {
                for (String key : data.keySet()) {
                    Object value = data.get(key);
                    if (value instanceof String) {
                        builder.getExtras().putString(key, (String) value);
                    } else if (value instanceof Double) {
                        builder.getExtras().putDouble(key, (Double) value);
                    } else if (value instanceof Boolean) {
                        builder.getExtras().putBoolean(key, (Boolean) value);
                    }
                }
            }
            
            notificationManager.notify(notificationId, builder.build());
            
        } catch (Exception e) {
            android.util.Log.e("LocalNotificationReceiver", "Failed to show scheduled notification", e);
        }
    }
    
    private int getNotificationIcon(Context context) {
        try {
            return context.getApplicationInfo().icon;
        } catch (Exception e) {
            return android.R.drawable.ic_dialog_info;
        }
    }
}

File: android/src/oldarch/kotlin/io/lokal/notifications/LocalNotificationReceiver.kt
package io.lokal.notifications

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.os.Bundle
import androidx.core.app.NotificationCompat
import androidx.core.app.NotificationManagerCompat

class LocalNotificationReceiver : BroadcastReceiver() {
    
    override fun onReceive(context: Context, intent: Intent) {
        val notificationId = intent.getIntExtra("notification_id", 0)
        val title = intent.getStringExtra("title") ?: ""
        val body = intent.getStringExtra("body") ?: ""
        val data = intent.getBundleExtra("data")

        // Create notification channel if needed
        createNotificationChannel(context)

        // Build notification
        val builder = NotificationCompat.Builder(context, "LocalNotifications")
            .setContentTitle(title)
            .setContentText(body)
            .setSmallIcon(getNotificationIcon(context))
            .setPriority(NotificationCompat.PRIORITY_DEFAULT)
            .setAutoCancel(true)

        // Show notification
        val notificationManager = NotificationManagerCompat.from(context)
        notificationManager.notify(notificationId, builder.build())
    }

    private fun createNotificationChannel(context: Context) {
        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.O) {
            val channel = android.app.NotificationChannel(
                "LocalNotifications",
                "Local Notifications",
                android.app.NotificationManager.IMPORTANCE_DEFAULT
            ).apply {
                description = "Local scheduled notifications"
            }

            val notificationManager = context.getSystemService(Context.NOTIFICATION_SERVICE) as android.app.NotificationManager
            notificationManager.createNotificationChannel(channel)
        }
    }

    private fun getNotificationIcon(context: Context): Int {
        return try {
            context.applicationInfo.icon
        } catch (e: Exception) {
            android.R.drawable.ic_dialog_info
        }
    }
}
